Literally.hs

```diff
@@ -49,6 +52,18 @@ instance FromType (n :: Type.Nat) Word.Word64 where
   type KnownType n Word.Word64 = (Type.KnownNat n, (Type.<=) n 18446744073709551615)
   fromType = fromInteger . fromType
 
+#if WORD_SIZE_IN_BITS == 32
+instance FromType (n :: Type.Nat) Word.Word where
+  type KnownType n Word.Word = (Type.KnownNat n, (Type.<=) n 4294967295)
```

Comment:
I would prefer if the only thing that differed between these instances was the upper bound, rather than repeating the entire instance. Could the upper bound be defined using CPP in relation to the `WORD_SIZE_IN_BITS`?